perf/trace.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM
+# Author: Kalpana Shetty <kalshett@in.ibm.com>
+#
+
+import os
+
+from avocado import Test
+from avocado.utils import distro, process, archive, git
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Linuxtrace(Test):
+
+    """
+    Linux Trace tests
+    """
+
+    @staticmethod
+    def run_cmd_out(cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True,
+                                     sudo=True)
```

Comment:
Output is not parsed to set the condition for failure. Please add this part.
If not your test case will always PASS irrespective of the actual test result.

Reply:
Okay, will incorporate this change.

Reply:
@nasastry Below is the code I will add with checking for failed case.
    def run_cmd_out(self, cmd):
        results = process.system_output(cmd, shell=True, sudo=True)
        match = re.search(br'fail', results, re.M | re.I)
        if match:
           self.log.info("Test failed: %s" % match)
        return results

Let me know if this would sufficient.